runtime/hsa-runtime/core/driver/xdna/amd_xdna_driver.cpp

```diff
@@ -60,27 +61,27 @@ namespace AMD {
 XdnaDriver::XdnaDriver(std::string devnode_name)
     : core::Driver(core::DriverType::XDNA, devnode_name) {}
 
-XdnaDriver::~XdnaDriver() { FreeDeviceHeap(); }
+XdnaDriver::~XdnaDriver() {}
 
-hsa_status_t XdnaDriver::DiscoverDriver() {
+hsa_status_t XdnaDriver::DiscoverDriver(std::unique_ptr<core::Driver> &driver) {
   const int max_minor_num(64);
   const std::string devnode_prefix("/dev/accel/accel");
 
   for (int i = 0; i < max_minor_num; ++i) {
-    std::unique_ptr<Driver> xdna_drv(
+    driver = std::unique_ptr<Driver>(
```

Comment:
Same comments here, you can have a temporary and only `std::move` if everything was successful and avoid `reset()` calls.

Reply:
Fixed